yarn-project/simulator/src/public/abstract_phase_manager.ts

```diff
@@ -515,29 +515,9 @@ function patchPublicStorageActionOrdering(
   // so the returned result will be a subset of the public kernel output.
 
   const simPublicDataReads = collectPublicDataReads(execResult);
-  // verify that each read is in the kernel output
```

Comment:
Removing the checks because the kernel is producing wrong leaf slots siloed with the first storage contract address for nested calls. Will fix this in another PR to include contract address for public reads and public writes, and silo them in the circuit.